starter_code/queries.md

```diff
@@ -4,80 +4,98 @@
 
-<!-- Your Code Goes Here -->
+{ partners: { \$exists: false } }
 
 ### 7. All the companies that have a null type of value on the `category_code` field.
 
-<!-- Your Code Goes Here -->
+{ category_code: { \$in: [null] } }
 
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
 
-<!-- Your Code Goes Here -->
+{ $and: [ { number_of_employees: { $lt: 1000}}, { number_of_employees: { \$gt: 100}}]}
+--> project field: {name: 1, number_of_employees: 1}
 
 ### 9. Order all the companies by their IPO price in a descending order.
 
-<!-- Your Code Goes Here -->
+--> sort field :{"ipo.valuation_amount": -1}
 
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
 
-<!-- Your Code Goes Here -->
+--> sort field: {number_of_employees: -1}
+--> limit : 10
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
-<!-- Your Code Goes Here -->
+{founded_month : {\$gt: 6}}
+--> limit: 1000
 
-<!-- ### 12. All the companies that have been 'deadpooled' after the third year. -->
+<!-- ### 12. All the companies that have been 'deadpooled' after the third year.
 
-<!-- Your Code Goes Here -->
+Your Code Goes Here -->
 
 ### 12. All the companies founded before 2000 that have an acquisition amount of more than 10.000.000
 
-<!-- Your Code Goes Here -->
+{ $and: [ { founded_year: { $lt: 2000}}, { "acquisition.price_amount" : { \$gt: 10000000}}]}
 
 ### 13. All the companies that have been acquired after 2010, order by the acquisition amount, and retrieve only their `name` and `acquisition` field.
 
-<!-- Your Code Goes Here -->
+{ "acquisition.acquired_year": {\$gt: 2010} }
+--> project : {name: 1, acquisition: 1}
+--> sort: {"acquisition.price_amount": 1}
 
 ### 14. Order the companies by their `founded year`, retrieving only their `name` and `founded year`.
 
-<!-- Your Code Goes Here -->
+--> project : {name: 1, founded_year: 1}
+--> sort : {founded_year: 1}
+
+<!-- (years with "null" value come first) -->
 
 ### 15. All the companies that have been founded on the first seven days of the month, including the seventh. Sort them by their `acquisition price` in a descending order. Limit the search to 10 documents.
 
-<!-- Your Code Goes Here -->
+{founded_day: {\$lte: 7}}
+--> sort: {"acquisition.price_amount": -1}
+--> limit: 10
 
 ### 16. All the companies on the 'web' `category` that have more than 4000 employees. Sort them by the amount of employees in ascending order.
 
-<!-- Your Code Goes Here -->
+{ $and: [ { category_code: { $in: ["web"]}}, {number_of_employees : { \$gt: 4000}} ] }
+--> sort: {number_of_employees: 1}
 
 ### 17. All the companies whose acquisition amount is more than 10.000.000, and currency is 'EUR'.
 
-<!-- Your Code Goes Here -->
+{ $and: [ {"acquisition.price_amount": {$gt: 10000000}}, {"acquisition.price_currency_code": { \$in: ["EUR"]} } ]}
 
```

Comment:
when you need a value equal to something you can write: {property: value} without using $in. $in will be useful to search if an array includes a value